lib/matplotlib/container.py

```diff
@@ -148,6 +148,66 @@ def __init__(self, lines, has_xerr=False, has_yerr=False, **kwargs):
         super().__init__(lines, **kwargs)
 
 
+class PieContainer:
+    """
+    Container for the artists of pie charts (e.g. created by `.Axes.pie`).
+
+    You can access the patches and further parameters by the attributes.
+
+    Attributes
+    ----------
+    patches : list of `~matplotlib.patches.Wedge`
+        The artists of the pie wedges.
+
+    values : tuple of int or float
+        The data that the pie is based on.
+
+    fracs : tuple of float
+        The fraction of the pie that each wedge represent.
+
+    texts : list of list of `~matplotlib.text.Text`
+        The artists of any labels on the pie wedges, except those added with the
+        *autotexts* parameter of `~.Axes.pie`.  Each inner list has one
+        text label per wedge.
+
+    autotexts : list of `~matplotlib.text.Text`
+        The artists of any labels added with the *autotexts* parameter of
+        `~.Axes.pie`.
+
+    """
+    def __init__(self, patches, values, normalize):
+        self.patches = patches
+        self.texts = []
+        self.autotexts = None
+        self._values = values
+        self._normalize = normalize
+
+    @property
+    def values(self):
+        return tuple(self._values)
+
+    @property
+    def fracs(self):
+        if self._normalize:
+            return tuple(self._values / self._values.sum())
+        else:
+            return tuple(self._values)
+
+    def remove(self):
+        """Remove all wedges and texts from the axes"""
+        for artist_list in self.patches, self.texts, self.autotexts:
+            for artist in cbook.flatten(artist_list):
+                artist.remove()
+
+    def __getitem__(self, key):
```

Comment:
I decided to use `__getitem__` for this rather than `__iter__` as suggested at https://github.com/matplotlib/matplotlib/pull/29152#issuecomment-3436676165 because there is [one test](https://github.com/matplotlib/matplotlib/blob/db83efff4d7d3849f8bffbd1f6cdfc43d74c9aea/lib/matplotlib/tests/test_axes.py#L6584-L6590) that uses indexing of the return value.  If we are doing that then it seems plausible that downstream might be too.